selftest/map-of-maps/main.go

```diff
@@ -0,0 +1,379 @@
+package main
+
+import "C"
+
+import (
+	"encoding/binary"
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+	"unsafe"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+)
+
+// This selftest deals with interactions involving hash maps and an array of maps.
+// It utilizes the functions CreateMap() and GetMapByID(), which internally call
+// bpf_map_create() and bpf_map_get_fd_by_id() - 'low-level' APIs provided by libbpf.
+// As such, it does not have access to the higher level APIs in libbpf, which are
+// denoted by starting with `bpf_map__*`.
```

Comment:
Too "chatGPT" for me. No need for this extra information (the first part is good IMO).

Reply:
It is a shortened rewrite of the initially proposed. But for sure, we can make it even shorter.